app/routes.py

```diff
@@ -1,2 +1,245 @@
+                "id": new_task.task_id,
+                "title": new_task.title,
+                "description": new_task.description,
+                "is_complete": True if new_task.completed_at else False
+            }
+        }, 201
+
+@tasks_bp.route("/<task_id>", methods=["GET", "DELETE", "PUT"])
+def handle_task(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response("", 404)
+
+    if request.method == "GET":
+        result = {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": True if task.completed_at else False
+            }
+        if task.goal_id:
+            result["goal_id"] = task.goal_id
+        return {
+            "task": result
+        }
+    elif request.method == "DELETE":
+        message = {"details": f"Task {task.task_id} \"{task.title}\" successfully deleted"}
+        db.session.delete(task)
+        db.session.commit()
+        return make_response(message)
+    elif request.method == "PUT":
+        form_data = request.get_json()
+
+        task.title = form_data["title"]
+        task.description = form_data["description"]
+        task.completed_at = form_data["completed_at"]
+
+        db.session.commit()
+
+        return make_response({
+                "task": {
+                    "id": task.task_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": True if task.completed_at else False
+                }
+        })
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def handle_task_complete(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response("", 404)
+
+    task.completed_at = datetime.utcnow()
+    db.session.commit()
+
+    response = requests.post('https://slack.com/api/chat.postMessage', params={'channel':'task-notifications', 'text': f'Someone just completed the task {task.title}'}, headers={'Authorization': os.environ.get("SLACKBOT_API_KEY")})
+
+    json_response = response.json()
+    print(json_response)
+
+    return make_response({
+            "task": {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": True
+            }
+    })
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def handle_task_incomplete(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response("", 404)
+
+    task.completed_at = None
+
+    db.session.commit()
+
+    return make_response({
+            "task": {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": False
+            }
+    })
+
+@goals_bp.route("", methods=["POST", "GET"])
+def handle_goals():
```

Comment:
Similar feedback about splitting these functions, and moving validation and dictionary-handling logic around that I made for `Task` will also apply to `Goals`. These are common themes for any model with CRUD operations.